shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/controllers/BridgeIngressController.java

```diff
@@ -67,6 +69,21 @@ public void init(EventSourceManager eventSourceManager) {
             bridgeIngress.setStatus(new BridgeIngressStatus(PhaseType.AUGMENTATION));
             return UpdateControl.updateStatusSubResource(bridgeIngress);
         }
+        LOGGER.info("Ingress deployment BridgeIngress: '{}' in namespace '{}' is ready", bridgeIngress.getMetadata().getName(), bridgeIngress.getMetadata().getNamespace());
+
+        // Create Service
+        Service service = bridgeIngressService.fetchOrCreateBridgeIngressService(bridgeIngress, deployment);
+        if (service.getStatus() == null) {
+            LOGGER.info("Ingress service BridgeIngress: '{}' in namespace '{}' is NOT ready", bridgeIngress.getMetadata().getName(),
+                    bridgeIngress.getMetadata().getNamespace());
+            bridgeIngress.setStatus(new BridgeIngressStatus(PhaseType.AUGMENTATION));
+            return UpdateControl.updateStatusSubResource(bridgeIngress);
+        }
+        LOGGER.info("Ingress service BridgeIngress: '{}' in namespace '{}' is ready", bridgeIngress.getMetadata().getName(), bridgeIngress.getMetadata().getNamespace());
```

Comment:
Are we sure that if `service.getStatus() != null` this implies that it is ready?

Reply:
Yes, I am :) If anybody has a proof that this is not always true just let me know

Reply:
cc @sutaakar ^^ ?

Reply:
I guess so. I am not aware of any dedicated attribute or annotation which describes Service status.